From 4bb0eddb89e3e5c305954bd2b66e82d277509dcf Mon Sep 17 00:00:00 2001
From: Andreas Mohr <and@gmx.li>
Date: Sat, 20 Feb 2016 13:16:40 +0000
Subject: [PATCH] editor/editcmd.c: cleanup some -Wcast-qual warnings

editcmd.c:2535:70: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
         disp1 = edit_replace_cmd__conv_to_display (saved1 ? saved1 : (char *) "");
                                                                      ^
editcmd.c:2536:70: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
         disp2 = edit_replace_cmd__conv_to_display (saved2 ? saved2 : (char *) "");
                                                                      ^
editcmd.c:2747:20: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
     WEdit *edit = ((edit_search_status_msg_t *) user_data)->edit;
                    ^

Signed-off-by: Andreas Mohr <and@gmx.li>
---
 lib/charsets.c       |  4 ++--
 lib/charsets.h       |  4 ++--
 src/editor/editcmd.c | 10 +++++-----
 3 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/lib/charsets.c b/lib/charsets.c
index 93d87b2..95c4f61 100644
--- a/lib/charsets.c
+++ b/lib/charsets.c
@@ -354,7 +354,7 @@ convert_to_display (char *str)
 /* --------------------------------------------------------------------------------------------- */
 
 GString *
-str_convert_to_display (char *str)
+str_convert_to_display (const char *str)
 {
     return str_nconvert_to_display (str, -1);
 
@@ -363,7 +363,7 @@ str_convert_to_display (char *str)
 /* --------------------------------------------------------------------------------------------- */
 
 GString *
-str_nconvert_to_display (char *str, int len)
+str_nconvert_to_display (const char *str, int len)
 {
     GString *buff;
     GIConv conv;
diff --git a/lib/charsets.h b/lib/charsets.h
index 2b95d8f..2e6b6c5 100644
--- a/lib/charsets.h
+++ b/lib/charsets.h
@@ -69,8 +69,8 @@ int convert_from_8bit_to_utf_c2 (const char input_char);
 GString *str_convert_to_input (char *str);
 GString *str_nconvert_to_input (char *str, int len);
 
-GString *str_convert_to_display (char *str);
-GString *str_nconvert_to_display (char *str, int len);
+GString *str_convert_to_display (const char *str);
+GString *str_nconvert_to_display (const char *str, int len);
 
 /*** inline functions ****************************************************************************/
 
diff --git a/src/editor/editcmd.c b/src/editor/editcmd.c
index 0d2d6e1..99c2803 100644
--- a/src/editor/editcmd.c
+++ b/src/editor/editcmd.c
@@ -909,7 +909,7 @@ editcmd_find (edit_search_status_msg_t * esm, gsize * len)
 /* --------------------------------------------------------------------------------------------- */
 
 static char *
-edit_replace_cmd__conv_to_display (char *str)
+edit_replace_cmd__conv_to_display (const char *str)
 {
 #ifdef HAVE_CHARSET
     GString *tmp;
@@ -2532,8 +2532,8 @@ edit_replace_cmd (WEdit * edit, int again)
     {
         char *tmp_inp1, *tmp_inp2;
 
-        disp1 = edit_replace_cmd__conv_to_display (saved1 ? saved1 : (char *) "");
-        disp2 = edit_replace_cmd__conv_to_display (saved2 ? saved2 : (char *) "");
+        disp1 = edit_replace_cmd__conv_to_display (saved1 ? saved1 : "");
+        disp2 = edit_replace_cmd__conv_to_display (saved2 ? saved2 : "");
 
         edit_push_undo_action (edit, KEY_PRESS + edit->start_display);
 
@@ -2744,7 +2744,7 @@ edit_replace_cmd (WEdit * edit, int again)
 mc_search_cbret_t
 edit_search_cmd_callback (const void *user_data, gsize char_offset, int *current_char)
 {
-    WEdit *edit = ((edit_search_status_msg_t *) user_data)->edit;
+    WEdit *edit = ((const edit_search_status_msg_t *) user_data)->edit;
 
     *current_char = edit_buffer_get_byte (&edit->buffer, (off_t) char_offset);
     return MC_SEARCH_CB_OK;
